File: mongo/csharp/delete/mongo_delete.cs
// ----------------------------------------------------------------
/*
	mongo_delete.cs

					Sep/02/2013

*/
// ----------------------------------------------------------------
using	System;
using	MongoDB;
// ----------------------------------------------------------------
public static class mongo_delete
{

// ----------------------------------------------------------------
public static void Main (string[] args)
{
	Console.WriteLine ("*** 開始 ***");

	string	key_in = args[0];
	Console.WriteLine (key_in);

	String db_name = "city_db";
	String col_name = "saitama";

	Mongo mm = new Mongo();
	mm.Connect();

	IMongoCollection things
		= mm.GetDatabase(db_name).GetCollection (col_name);

	Document doc_del = new Document ();
	doc_del["key"] = key_in;

	things.Remove (doc_del);

	mongo_manipulate.mongo_display_proc (things);

	mm.Disconnect();

	Console.WriteLine ("*** 終了 ***");
}

// ----------------------------------------------------------------
}
// ----------------------------------------------------------------

File: mongo/fsharp/delete/mongo_delete.fs
// ----------------------------------------------------------------
//
//	mongo_delete.fs
//
//					Dec/23/2011
//
//
// ----------------------------------------------------------------
open System
open MongoDB
// ----------------------------------------------------------------
[<EntryPoint>]
let main (args : string[] ) =

  printfn "*** 開始 ***"
  let key_in = args.[0]

  printfn "%s" key_in

  let db_name = "city_db"
  let col_name = "saitama"

  let mm = new Mongo()
  mm.Connect()

  let things = mm.GetDatabase(db_name).GetCollection (col_name)

  let doc_del = new Document ()
  let doc_del["key"] = key_in

  things.Remove (doc_del)

  mongo_manipulate.mongo_display_proc (things)

  mm.Disconnect()

  printfn "*** 終了 ***"
  0

// ----------------------------------------------------------------
